### ISSUE TITLE
LFW verification accuracy lower than expected

### ISSUE THREAD

**Issue by zack-kimble** [timesler/facenet-pytorch#120]
I only get around ~96.3% accuracy when using both MTCNN and the pretrained vgg model to perform face verification on LFW. I'm using the functions for evaluation from `davidsandberg/facenet` which follow the protocol laid out by LFW:http://vis-www.cs.umass.edu/lfw/#views.

When I switch to using the pretrained model with cropped images produced by the `davidsandberg/facenet` MTCNN, I get 99.3% accuracy. So it seems like there is something about the pytorch MTCNN that causes a disconnect with the pretrained model. I've tried multiple settings of the margin on MTCNN, but still end up around 96%. Is there something I'm missing?

I've uploaded a minimal notebook to illustrate this here: https://github.com/zack-kimble/cv_sandbox/blob/master/facenet/facenet_pytorch_lfw_verification.ipynb

**Comment by zack-kimble** [comment]
Figured it out. When selecting a single face from an image, Sandberg's MTCNN implementation evaluates each bounding boxes size _and_ offset from center.
`index = np.argmax(bounding_box_size-offset_dist_squared*2.0)`

This results in better crops for LFW. The pytorch MTCNN ends up selecting a lot of large faces on the margins of the picture, which greatly increases false negatives. My suspicion is that this is generally true for any image where only one face is being selected. Photographers attempt to center.

I will add that option to MTCNN and submit a PR.

**Comment by timesler** [comment]
Closed by #122
